core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -181,9 +181,15 @@ public static function indexPage($arrData)
 								->limit(1)
 								->execute($arrSet['checksum'], $arrSet['pid']);
 
-		// Update the URL if the new URL is shorter or the current URL is not canonical
-		if ($objIndex->numRows && $objIndex->url != $arrSet['url'])
+		if ($objIndex->numRows)
 		{
+			// The page has already been indexed and has not changed (see #2235)
+			if ($objIndex->url == $arrSet['url'])
+			{
+				return true;
```

Comment:
```suggestion
				return false;
```
The doccomment says `@return boolean True if a new record was created`

Reply:
Good catch. 👍🏻